drivers/usbdev_synopsys_dwc2/usbdev_synopsys_dwc2.c

```diff
@@ -578,15 +580,41 @@ static void _sleep_periph(const dwc2_usb_otg_fshs_config_t *conf)
     /* Unblocking STM32_PM_STOP during suspend on the stm32f446 breaks
      * while (un)blocking works on the stm32f401, needs more
      * investigation with a larger set of chips */
-#ifdef STM32_USB_OTG_CID_1x
+#if defined(STM32_USB_OTG_CID_1x)
     pm_unblock(STM32_PM_STOP);
+#elif defined(MCU_EFM32)
+    /* switch USB core clock source either to LFXO or LFRCO */
+#if defined(CPU_FAM_EFM32GG12B)
+    CMU->USBCTRL &= ~_CMU_USBCTRL_USBCLKSEL_MASK;
+    CMU->USBCTRL |= (CLOCK_LFA == cmuSelect_LFXO) ? CMU_USBCTRL_USBCLKSEL_LFXO
+                                                  : CMU_USBCTRL_USBCLKSEL_LFRCO;
+#elif defined(CPU_FAM_EFM32GG) || defined(CPU_FAM_EFM32LG)
+    CMU->CMD &= ~_CMU_CMD_USBCCLKSEL_MASK;
+    CMU->CMD |= (CLOCK_LFA == cmuSelect_LFXO) ? CMU_CMD_USBCCLKSEL_LFXO
+                                              : CMU_CMD_USBCCLKSEL_LFRCO;
+#else
+#error "EFM32 family not yet supported"
+#endif
```

Comment:
The emlib provided by the `gecko_sdk` has an helper for clock source selection. Without having this tested, this might work for your CPU and probably will work for future CPUs, too.

```suggestion
    CMU_ClockSelectSet(cmuClock_USB, CLOCK_LFA);
```

Reply:
After having a look into the implementation of this function, it should work. I will test it for EFM32GG12B, but I can't test it for the other EFM32 variants.